feat: add GAR to cicd pipeline

rutalreja-deloitte opened the pull request:
Adds Cloud builds submit for skaffold_image.yaml
Modularize app_cicd TF code
Adds new 5th repo boa_source repo
Trims down to only one trigger on boa_source repo
Changes GCR to GAR

bharathkkb commented on 5-infrastructure/modules/app_cicd_pipeline/repo.tf:
```diff
@@ -55,22 +55,37 @@ resource "google_storage_bucket_iam_member" "cloudbuild_artifacts_iam" {
-    _ARTIFACT_BUCKET_NAME = google_storage_bucket.cloudbuild_artifacts["${each.value}-ab"].name
+    _GAR_REPOSITORY    = local.gar_name
+    _DEFAULT_REGION    = var.primary_location
+    _CACHE_BUCKET_NAME = "${var.app_cicd_project_id}_cloudbuild"
```
We should get the bucket name from the resource `google_storage_bucket. cache_bucket` so the dependency can be set correctly

bharathkkb commented on build_app/README.md:
```diff
@@ -4,31 +4,11 @@ This is an opinionated repository demonstrating Cloud Build based builds of Bank
-| gsutil | Interact with Google Cloud Storage buckets | $> `gcloud components install gsutil` |
+1. Clone Bank of Anthos Repo into `bank-of-anthos-source` Cloud Source Repo in project `prj-bu1-s-app-cicd-xxxx` under folder `fldr-common`
+1. Copy in files `cloudbuild-build-boa.yaml` and `policies` folder into above repo
+1. Run `sed -i.bak "s|gcr.io/bank-of-anthos|$_DEFAULT_REGION-docker.pkg.dev/$PROJECT_ID/$_GAR_REPOSITORY|g" skaffold.yaml` and `sed -i.bak "s|gitCommit: {}|sha256: {}|g" skaffold.yaml` at the root folder level while replacing the variables with the outputs obtained from runnning 5-infrastructure/business-unit-1/shared
```
lets keep this as a codeblock

```
sed -i.bak "s|gcr.io/bank-of-anthos|$_DEFAULT_REGION-docker.pkg.dev/$PROJECT_ID/$_GAR_REPOSITORY|g" skaffold.yaml
sed -i.bak "s|gitCommit: {}|sha256: {}|g" skaffold.yaml
```

bharathkkb commented on 5-infrastructure/business_unit_1/shared/README.md:
```diff
@@ -16,5 +13,6 @@
 |------|-------------|
 | bin\_auth\_attestor\_names | Names of Attestors |
 | bin\_auth\_attestor\_project\_id | Project ID where attestors get created |
+| boa\_artifact\_repo | GAR Repo created to store runner images |
```
```suggestion
| boa\_artifact\_repo | GAR Repo created to store BoA images |
```

rutalreja-deloitte commented:
Added Permissions for Cloud Armor and External IP creation in 4-projects SA
Changed to random_string from random_id on keyring name in 5-infra/shared
Root README updated
build_app/README Images Updated

bharathkkb commented on 5-infrastructure/modules/app_cicd_pipeline/repo.tf:
```diff
@@ -55,22 +55,37 @@ resource "google_storage_bucket_iam_member" "cloudbuild_artifacts_iam" {
   trigger_template {
     branch_name = ".*"
-    repo_name   = each.value
+    repo_name   = "bank-of-anthos-source"
```
we should make this a var in the module and hardcode to `bank-of-anthos-source` in the root module

rutalreja-deloitte commented on 6-anthos-install/acm-repos/accounts/contacts.yaml:
```diff
@@ -31,7 +31,7 @@ spec:
       containers:
         - name: contacts
-          image: gcr.io/prj-bu1-d-boa-gke-ecb0/contacts@sha256:bddc2b1bae8e77605fbda4997382f5e8e320a0b89f42d5dfbfe99d441ed5f1ff
+          image: <Region - us-east1>-docker.pkg.dev/<Project ID - prj-bu1-s-app-cicd-xxxx>/<GAR_REPOSITORY - prj-bu1-s-app-cicd-xxxx-boa-image-repo>/contacts@sha256:bddc2b1bae8e77605fbda4997382f5e8e320a0b89f42d5dfbfe99d441ed5f1ff
```
@bharathkkb so if it is `gcr.io/project_id` or `image: gcr.io/bank-of-anthos/contacts:v0.4.3` do I still update it or let it be
